src/study.py

```diff
@@ -335,7 +337,12 @@ def save_model(self):
                                 print("FAILED TO SAVE MODEL {} DUE TO FOLLOW EXCEPTION".format(study.name))
                                 print(e)
         
+                    # free up memory again
+                    metric = str(study.metric)
+                    del study.model
+                    torch.cuda.empty_cache()
```

Comment:
No need to hold on to models after study is complete. Was sometimes causing OOM errors on slurm-based servers. This ensures to free up some of that space again to protect against unneeded failures.